plugins/gateway/kong/plugin.go

```diff
@@ -113,11 +113,11 @@ func shouldTrace(kong *pdk.PDK) (bool, error) {
 	if err != nil {
 		return false, fmt.Errorf("failed to get routed service: %v", err)
 	}
-	host, _ := parseHost(routedService.Host)
-	if traceSamplingClient.ShouldTrace(host) {
+	host, port, _ := parseKongHost(routedService.Host)
```

Comment:
Not sure. 
We use kong.Router.GetService().Host(), which the docs state:
`// kong.Router.GetService() returns the current service entity.
// The request will be targetted to this upstream service.`

On the other hand there is no direct access to the URL, just to parts of the URL via:
 kong.Request.GetForwardedPort() and kong.Request.GetForwardedHost()
 
 I will check these to see if it is working and maybe use them, as it seems more correct (again, not sure)

Reply:
I think it is best to take the host and port from the routing information after all, since these are for sure the target service taken from the Ingress resource